storage/src/main/java/org/apache/kafka/storage/internals/log/LazyIndex.java

```diff
@@ -166,20 +166,25 @@ public File file() {
 
     @SuppressWarnings("unchecked")
     public T get() throws IOException {
-        if (indexWrapper instanceof IndexValue<?>)
-            return ((IndexValue<T>) indexWrapper).index;
-        else if (indexWrapper instanceof IndexFile) {
+        IndexWrapper wrapper = indexWrapper;
+        if (wrapper instanceof IndexValue<?>)
+            return ((IndexValue<T>) wrapper).index;
```

Comment:
This change is not strictly necessary with the way the code behaves today (we never change from `IndexValue` to something else), but it's cleaner not to access the field again given that we are doing it without a lock here.